### PR TITLE
MTDSA-14097 RAML to OAS

### PR DESCRIPTION
**Pull request by mattstephens-hmrc**

### CODE REVIEW

File: resources/public/api/conf/2.0/uk_property_period_summary_create.yaml [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+            FORMAT_TAX_YEAR:
+              $ref: './common/errors.yaml#/components/examples/formatTaxYear' 
+            FORMAT_FROM_DATE:
+              $ref: './common/errors.yaml#/components/examples/formatFromDate' 
+            FORMAT_TO_DATE:
+              $ref: './common/errors.yaml#/components/examples/formatToDate' 
+            FORMAT_VALUE:
+              $ref: './common/errors.yaml#/components/examples/formatValue' 
+            RULE_INVALID_SUBMISSION_END_DATE:
+              $ref: './common/errors.yaml#/components/examples/invalidSubmissionEndDate'
```
Comment (QHMRC) [inline resources/public/api/conf/2.0/uk_property_period_summary_create.yaml:113]: Maybe you need to double check. This ones not in the RAML

File: resources/public/api/conf/2.0/uk_property_period_summary_create.yaml [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+            RULE_INVALID_SUBMISSION_PERIOD:
+              $ref: './common/errors.yaml#/components/examples/invalidSubmissionPeriod'
```
Comment (QHMRC) [inline resources/public/api/conf/2.0/uk_property_period_summary_create.yaml:115]: Maybe you need to double check. This ones not in the RAML

File: resources/public/api/conf/2.0/historic_uk_fhl_annual_submission_retrieve.yaml [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+      </tbody>
+  </table>  
+
+tags:
+  - Historic FHL UK Property Business Annual Submission
+
+parameters:
+  - $ref: './common/pathParameters.yaml#/components/parameters/nino'
+  - $ref: './common/pathParameters.yaml#/components/parameters/taxYear'
+  - $ref: './common/headers.yaml#/components/parameters/acceptHeader'
```
Comment (QHMRC) [inline resources/public/api/conf/2.0/historic_uk_fhl_annual_submission_retrieve.yaml:34]: the raml version of this doesnt contain an acceptHeader. Not sure if needed

File: resources/public/api/conf/2.0/historic_uk_fhl_period_summary_create.yaml [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+          value:
+            $ref: './examples/historic_uk_fhl_period_summary_create/request.json'
+        consolidated-expense-request:
+          description: Consolidated Expenses request
+          value:
+            $ref: './examples/historic_uk_fhl_period_summary_create/consolidated_request.json'
+
+responses:
+  "201":
+    description: Success
```
Comment (QHMRC) [inline resources/public/api/conf/2.0/historic_uk_fhl_period_summary_create.yaml:77]: Success (created) ?

File: resources/public/api/conf/2.0/historic_uk_fhl_period_summary_retrieve.yaml [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+  <table>
+      <thead>
+          <tr>
+              <th>Header Value (Gov-Test-Scenario)</th>
+              <th>Scenario</th>
+          </tr>
+      </thead>
+      <tbody>
+          <tr>
+              <td><p>N/A - DEFAULT</p></td>
```
Comment (QHMRC) [inline resources/public/api/conf/2.0/historic_uk_fhl_period_summary_retrieve.yaml:16]: possibly wrong scenarios added

File: resources/public/api/conf/2.0/historic_uk_fhl_period_summary_amend.yaml [hunk 6]
```diff
@@ -0,0 +1,122 @@
...
+          FORMAT_VALUE:
+            $ref: './common/errors.yaml#/components/examples/formatFhlValue' 
+          RULE_TAX_YEAR_RANGE_INVALID:
+            $ref: './common/errors.yaml#/components/examples/ruleTaxYearRangeInvalid' 
+          RULE_TAX_YEAR_NOT_SUPPORTED:
+            $ref: './common/errors.yaml#/components/examples/ruleTaxYearNotSupported'
+          RULE_BOTH_EXPENSES_SUPPLIED:
+            $ref: './common/errors.yaml#/components/examples/ruleBothExpensesSupplied' 
+          RULE_TYPE_OF_BUSINESS_INCORRECT:
+            $ref: './common/errors.yaml#/components/examples/ruleTypeOfBusinessIncorrectUk'
```
Comment (QHMRC) [inline resources/public/api/conf/2.0/historic_uk_fhl_period_summary_amend.yaml:97]: ruleTypeOfBusinessIncorrectUk not required

File: resources/public/api/conf/2.0/historic_uk_fhl_period_summary_amend.yaml [hunk 7]
```diff
@@ -0,0 +1,122 @@
...
+      application/json:
+        schema:
+          $ref: './schemas/error-response.yaml'
+        examples:
+          FORMAT_NINO:
+            $ref: './common/errors.yaml#/components/examples/formatNino'
+          FORMAT_VALUE:
+            $ref: './common/errors.yaml#/components/examples/formatFhlValue' 
+          RULE_TAX_YEAR_RANGE_INVALID:
+            $ref: './common/errors.yaml#/components/examples/ruleTaxYearRangeInvalid'
```
Comment (QHMRC) [inline resources/public/api/conf/2.0/historic_uk_fhl_period_summary_amend.yaml:91]: not requried

File: resources/public/api/conf/2.0/historic_uk_non_fhl_period_summary_create.yaml [hunk 8]
```diff
@@ -0,0 +1,141 @@
...
+          <tr>
+              <td><p>N/A - DEFAULT</p></td>
+              <td><p>Simulate success response.</p></td>
+          </tr>
+          <tr>
+              <td><p>NOT_FOUND</p></td>
+              <td><p>Simulates the scenario where the given resource could not be found.</p></td>
+          </tr>
+          <tr>
+              <td><p>OVERLAPPING</p></td>
```
Comment (QHMRC) [inline resources/public/api/conf/2.0/historic_uk_non_fhl_period_summary_create.yaml:24]: double check the scenarios file

File: resources/public/api/conf/2.0/historic_uk_non_fhl_period_summary_retrieve.yaml [hunk 9]
```diff
@@ -0,0 +1,100 @@
...
+      </tbody>
+  </table>
+
+tags:
+  - Historic non-FHL UK Property Income & Expenses Period Summary
+
+parameters:
+  - $ref: './common/pathParameters.yaml#/components/parameters/nino'
+  - $ref: './common/pathParameters.yaml#/components/parameters/periodId'
+  - $ref: './common/headers.yaml#/components/parameters/acceptHeader'
```
Comment (QHMRC) [inline resources/public/api/conf/2.0/historic_uk_non_fhl_period_summary_retrieve.yaml:41]: missing content-type

File: resources/public/api/conf/2.0/historic_uk_fhl_period_summary_retrieve.yaml [hunk 10]
```diff
@@ -0,0 +1,121 @@
...
+            value:
+              $ref: './examples/uk_property_period_summary_retrieve/consolidated_response.json'
+  "400":
+    description: Bad request
+    content:
+      application/json:
+        schema:
+          $ref: './schemas/error-response.yaml'
+        examples:
+          FORMAT_NINO:
```
Comment (QHMRC) [inline resources/public/api/conf/2.0/historic_uk_fhl_period_summary_retrieve.yaml:86]: double check 400 error codes. Seems to have more error codes than the RAML

File: resources/public/api/conf/2.0/foreign_property_period_summary_create.yaml [hunk 11]
```diff
@@ -0,0 +1,162 @@
...
+            value:
+              $ref: './examples/foreign_property_period_summary_create/request.json'
+          consolidated-expenditure-request:
+            description: Consolidated Expenditure request
+            value:
+              $ref: './examples/foreign_property_period_summary_create/consolidated_request.json'
+
+  responses:
+    "201":
+      description: Success
```
Comment (QHMRC) [inline resources/public/api/conf/2.0/foreign_property_period_summary_create.yaml:84]: Not sure whats our standard.
Maybe `Success (Created)` or `Created`?

File: resources/public/api/conf/2.0/historic_uk_fhl_annual_submission_retrieve.yaml (hunk 3 above)
Comment (mattstephens-hmrc) [reply resources/public/api/conf/2.0/historic_uk_fhl_annual_submission_retrieve.yaml:34]: It's needed

File: resources/public/api/conf/2.0/historic_uk_fhl_annual_submission_delete.yaml [hunk 12]
```diff
@@ -0,0 +1,85 @@
...
+  - $ref: './common/headers.yaml#/components/parameters/authorizationWriteScopeHeader'
+  - $ref: './common/headers.yaml#/components/parameters/testScenarioHeader'
+
+security:
+  - User-Restricted:
+      - write:self-assessment
+
+responses:
+  "204":
+    description: Success(No Content)
```
[the remark this answers is not in the document]
Comment (mattstephens-hmrc) [reply resources/public/api/conf/2.0/historic_uk_fhl_annual_submission_delete.yaml:44]: We're not really fixing the RAML descriptions here, there'll be follow-up tickets for that later

File: resources/public/api/conf/2.0/historic_uk_fhl_period_summary_create.yaml (hunk 4 above)
Comment (mattstephens-hmrc) [reply resources/public/api/conf/2.0/historic_uk_fhl_period_summary_create.yaml:77]: same (not fixing-up the RAML for now)

File: resources/public/api/conf/2.0/historic_uk_non_fhl_period_summary_retrieve.yaml (hunk 9 above)
Comment (mattstephens-hmrc) [reply resources/public/api/conf/2.0/historic_uk_non_fhl_period_summary_retrieve.yaml:41]: not needed for a retrieve

File: resources/public/api/conf/2.0/foreign_property_period_summary_create.yaml (hunk 11 above)
Comment (mattstephens-hmrc) [reply resources/public/api/conf/2.0/foreign_property_period_summary_create.yaml:84]: same (not fixing-up the RAML for now)

**Review by QHMRC — APPROVED** [review]

**Pull request merged**